### ISSUE TITLE
Custom scalars should work without having to add `@specifiedBy`

### ISSUE THREAD

**Issue by guidomb** [apollographql/apollo-rs#572]
## Description

Currently if the schema being parsed by Apollo compiler includes custom scalar definition without the `@specifiedBy` directive the schema won't compile. In the [GraphQL specification (revision October 2021) ](https://spec.graphql.org/October2021/#sec-Scalars.Custom-Scalars) they use the word SHOULD when referring to custom scalars having to add the `@specifiedBy` directive.

> When defining a custom scalar, GraphQL services should provide a [scalar specification URL](https://spec.graphql.org/October2021/#scalar-specification-url) via the @specifiedBy directive or the specifiedByURL introspection field. This URL must link to a human-readable specification of the data format, serialization, and coercion rules for the scalar.

My understanding is that custom scalar without including the `@specifiedBy` directive should work. Am I correct?

Looks like this rule has been added in https://github.com/apollographql/apollo-rs/pull/234

### Steps to reproduce

```graphql
scalar DateTime
type Object {
  field: DateTime
}
```

### Expected result

It should compile

### Actual result

It does not compile because it requires that `DateTime` to be definition to include `@specifiedBy`

### Environment

- Operating system and version: macOS 13.3.1
- Shell (bash/zsh/powershell): zsh
- `apollo-rs` crate: 0.8.0
- Crate version: 0.8.0

**Comment by guidomb** [comment]
@lrlna cool! In my particular case I ended-up working around this by injecting the `@specifiedBy` directive before passing the schema to `ApolloCompiler`, I don't control the schema's my app has to handle and that was a quick hack that worked.

Funny thing is that I found this issue with a schema that was generated from an introspection query using https://github.com/apollographql/introspector-gadget. The original schema did specify the directive for each custom scalar that introspector didn't support asking for the directive and therefore not including it in the generated schema SDL. That one was an easy fix https://github.com/apollographql/introspector-gadget/pull/9.

**Comment by SimonSapin** [comment]
I believe this is fixed in apollo-compiler 1.0, now in beta. The new signature for schema validation is:

https://docs.rs/apollo-compiler/1.0.0-beta.1/apollo_compiler/schema/struct.Schema.html#method.validate
```rust
impl Schema {
    pub fn validate(&self) -> Result<Diagnostics, Diagnostics> { .. }
}
```

The result will be `Err` if there are any error-level diagnostics (meaning the schema is invalid). `Result::Ok` on other hand indicates that the schema is valid. It contains a set of `Diagnostics` that’s either empty or only has advice-level or warning-level thing like missing `@specifiedBy`.
